semialign/src/Data/Semialign/Internal.hs

```diff
@@ -481,12 +484,14 @@ instance SemialignWithIndex Int NonEmpty
 instance ZipWithIndex Int NonEmpty
 instance RepeatWithIndex Int NonEmpty
 
+#if !MIN_VERSION_base(4,16,0)
 deriving instance Semialign Option
 deriving instance Align Option
 deriving instance Unalign Option
 deriving instance Zip Option
 deriving instance Repeat Option
 deriving instance Unzip Option
+#endif
 
 {-
 deriving instance SemialignWithIndex () Option
```

Comment:
Do you want to move the `#endif` to below this block comment, so that if it's ever uncommented it will do the right thing?